ethers/providers/jsonrpc.nim

```diff
@@ -88,10 +88,16 @@ proc new*(
 proc callImpl(
   client: RpcClient,
   call: string,
-  args: JsonNode): Future[JsonNode] {.async: (raises: [JsonRpcProviderError]).} =
+  args: JsonNode): Future[JsonNode] {.async: (raises: [CancelledError, JsonRpcProviderError]).} =
+
+  var response: JsonString
+  try:
+    response = await client.call(call, %args)
+  except CancelledError as e:
+    raise e
+  except CatchableError as e:
+    raiseJsonRpcProviderError e.msg
```

Comment:
I would suggest properly wrap the error with keeping not only parent's message but the whole parent error in order to keep the stacktrace.